specutils/io/default_loaders/wcs_fits.py

```diff
@@ -86,3 +91,406 @@ def wcs1d_fits_writer(spectrum, file_name, **kwargs):
+
+    This method read the appropriate keywords. Calls the method _set_math_model
+    which decides what is the appropriate mathematical model to be used and
+    creates an then evaluates the model for an array.
+
+    Parameters
+    ----------
+
+    wcs : `~astropy.wcs.WCS`
+        Contains wcs information extracted from the header
+
+    dc_flag : int
+        Extracted from the header under the keyword DC-FLAG which defines what
+        kind of solution is described. For linear solutions it is 0 or 1.
+
+    Returns
+    -------
+
+    spectral_axis : `~numpy.ndarray`
+        Mathematical model of wavelength solution evluated for each pixel
+        position
+
+    """
+    wcs_dict = {'crval': wcs.wcs.crval[0],
+                'crpix': wcs.wcs.crpix[0],
+                'cdelt': wcs.wcs.cd[0],
+                'dtype': dc_flag,
+                'pnum': wcs._naxis[0]}
+
+    math_model = _set_math_model(wcs_dict=wcs_dict)
+
+    spectral_axis = math_model(range(wcs_dict['pnum']))
+
+    return spectral_axis
+
+
+def _read_non_linear_iraf_wcs(header, wcsdim):
+    """Read non-linear wavelength solutions written by IRAF
+
+    Extracts the appropriate information and organize it in a dictionary for
+    calling the method _set_math_model which decides what is the appropriate
+    mathematical model to be used according the the type of wavelength solution
+    it is dealing with.
+
+    Parameters
+    ----------
+
+    header : `~astropy.io.fits.header.Header`
+        Full header of file being loaded
+
+    wcsdim : int
+        Number of the wcs dimension to be read.
+
+     Returns
+    -------
+
+    spectral_axis : `~numpy.ndarray`
+        Mathematical model of wavelength solution evluated for each pixel
+        position
+    """
+
+    wat_wcs_dict = {}
+    ctypen = header['CTYPE{:d}'.format(wcsdim)]
+    logging.info('Attempting to read CTYPE{:d}: {:s}'.format(wcsdim, ctypen))
+    if ctypen == 'MULTISPE':
+        # TODO (simon): What is the * (asterisc) doing here?.
+        wat_head = header['WAT{:d}*'.format(wcsdim)]
+        if len(wat_head) == 1:
+            logging.debug('Get units')
+            wat_array = wat_head[0].split(' ')
+            for pair in wat_array:
+                split_pair = pair.split('=')
+                wat_wcs_dict[split_pair[0]] = split_pair[1]
+                # print(wat_head[0].split(' '))
+        elif len(wat_head) > 1:
+            wat_string = ''
+            for key in wat_head:
+                wat_string += header[key]
+            wat_array = shlex.split(wat_string.replace('=', ' '))
+            if len(wat_array) % 2 == 0:
+                for i in range(0, len(wat_array), 2):
+                    # if wat_array[i] not in wcs_dict.keys():
+                    wat_wcs_dict[wat_array[i]] = wat_array[i + 1]
+                    # print(wat_array[i], wat_array[i + 1])
+
+    for key in wat_wcs_dict.keys():
+        logging.info("{:d} -{:s}- {:s}".format(wcsdim,
```

Comment:
Is this too verbose on the command line? May want to use `logging.debug` so that it only logs all the these things when explicitly told to.

Reply:
Fixes